File: db4oj/core/src/com/db4o/internal/marshall/SlotFormat2.java
/* Copyright (C) 2008  Versant Inc.  http://www.db4o.com */

package com.db4o.internal.marshall;

import com.db4o.internal.*;
import com.db4o.typehandlers.*;


/**
 * @exclude
 */
public class SlotFormat2 extends SlotFormat {

    protected int handlerVersion() {
        return 2;
    }

    public boolean isIndirectedWithinSlot(TypeHandler4 handler) {
        return Handlers4.isVariableLength(handler);
    }

}

File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Internal/Marshall/SlotFormat2.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.Internal;
using Db4objects.Db4o.Internal.Marshall;
using Db4objects.Db4o.Typehandlers;

namespace Db4objects.Db4o.Internal.Marshall
{
	/// <exclude></exclude>
	public class SlotFormat2 : SlotFormat
	{
		protected override int HandlerVersion()
		{
			return 2;
		}

		public override bool IsIndirectedWithinSlot(ITypeHandler4 handler)
		{
			return Handlers4.IsVariableLength(handler);
		}
	}
}
